### PR TITLE
stubs, generator and config

### PR DESCRIPTION
**Pull request by koriaking**
remove soft delete cast https://github.com/laravel/framework/commit/b8d4869807185d6a5d4601bf0c87322474543eed

fix and add phpdocs
set default Illuminate\Database\Eloquent\Model

### CODE REVIEW

File: src/Generators/ModelGenerator.php [hunk 1]
```diff
@@ -101,18 +101,12 @@ private function fillSoftDeletes($templateData)
...
                 '$SOFT_DELETE_IMPORT$', "use Illuminate\\Database\\Eloquent\\SoftDeletes;\n",
                 $templateData
             );
-            $templateData = str_replace('$SOFT_DELETE$', infy_tab()."use SoftDeletes;\n", $templateData);
-            $deletedAtTimestamp = config('infyom.laravel_generator.timestamps.deleted_at', 'deleted_at');
-            $templateData = str_replace(
-                '$SOFT_DELETE_DATES$', infy_nl_tab()."protected \$dates = ['".$deletedAtTimestamp."'];\n",
-                $templateData
-            );
+            $templateData = str_replace('$SOFT_DELETE$', "use SoftDeletes;\n", $templateData);
```
Comment (ajayinfyom) [inline src/Generators/ModelGenerator.php]: we are given support to add a custom field for soft delete instant of delete_at only

File: src/Generators/ModelGenerator.php [hunk 2]
```diff
@@ -165,29 +159,48 @@ private function getPHPDocType($db_type, $relation = null, $relationText = null)
...
-                return '\\'.$this->commandData->config->nsModel.'\\'.$relation->inputs[0].' '.Str::camel($relationName);
+                return '\\' . $this->commandData->config->nsModel . '\\' . $relation->inputs[0] . ' ' . Str::camel($relationName);
             case '1tm':
             case 'mtm':
             case 'hmt':
-                return '\Illuminate\Database\Eloquent\Collection'.' '.Str::camel(Str::plural($relationText));
+                return '\Illuminate\Database\Eloquent\Collection' . ' ' . Str::camel(Str::plural($relationText));
             default:
-                $fieldData = SwaggerGenerator::getFieldType($db_type);
-                if (!empty($fieldData['fieldType'])) {
```
Comment (ajayinfyom) [inline src/Generators/ModelGenerator.php]: here in the default case, checking all fields types which are you added in the above cases

**Review by ajayinfyom — CHANGES_REQUESTED** [review]

File: src/Generators/ModelGenerator.php (hunk 1 above)
Comment (yevhenkoriakin) [reply src/Generators/ModelGenerator.php]: Yes, it`s work.  $SOFT_DELETE_DATES$ is unnecessary. I only removed the type cast.

File: src/Generators/ModelGenerator.php (hunk 2 above)
Comment (yevhenkoriakin) [reply src/Generators/ModelGenerator.php]: No, the decimal field in the table returns a "number" type.
https://docs.phpdoc.org/latest/guides/types.html
We should get "float"

Comment (yevhenkoriakin) [reply src/Generators/ModelGenerator.php]: public function initializeSoftDeletes()
    {
        $this->dates[] = $this->getDeletedAtColumn();
    }

https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/SoftDeletes.php#L34

File: src/Generators/ModelGenerator.php (hunk 1 above)
Comment (yevhenkoriakin) [reply src/Generators/ModelGenerator.php]: public function initializeSoftDeletes()
    {
        $this->dates[] = $this->getDeletedAtColumn();
    }

https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/SoftDeletes.php#L34

File: src/Generators/ModelGenerator.php (hunk 2 above)
Comment (ajayinfyom) [reply src/Generators/ModelGenerator.php]: ok, I will check it.

File: src/Generators/ModelGenerator.php (hunk 1 above)
Comment (ajayinfyom) [reply src/Generators/ModelGenerator.php]: sry, it does not work a custom field without declaring it.

if we use soft delete with a custom key than  must be declared in model

File: src/Generators/ModelGenerator.php (hunk 2 above)
Comment (ajayinfyom) [reply src/Generators/ModelGenerator.php]: I checked it, so here you can update in `SwaggerGenerator::getFieldType()` method of `SwaggerGenerator.php` as per need instant of added all cases.

Comment (ajayinfyom) [reply src/Generators/ModelGenerator.php]: I checked it.  here you can just update `SwaggerGenerator::getFieldType()` method of `SwaggerGenerator.php` as per needed instant of added all cases, already checked all cases in that function.

**Pull request closed without merging**